libsolidity/ast/Types.cpp

```diff
@@ -2025,7 +2025,7 @@ unsigned EnumType::memberValue(ASTString const& _member) const
 			return index;
 		++index;
 	}
-	BOOST_THROW_EXCEPTION(m_enum.createTypeError("Requested unknown enum value ." + _member));
+	solAssert(false, "Requested unknown enum value " + _member);
```

Comment:
This code was only ever called from within codegen and therefore the `TypeError` was not really utilised.